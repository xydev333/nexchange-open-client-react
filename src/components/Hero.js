import React, { Component } from 'react';
import { I18n, Trans } from 'react-i18next';

import ExchangeWidget from '../containers/ExchangeWidget';
import CoinPrices from '../containers/CoinPrices';
import ErrorAlert from '../components/ErrorAlert';

let scrollToElement;

class Hero extends Component {
  componentDidMount() {
    scrollToElement = require('scroll-to-element');
  }

  render() {
    return (
	<I18n ns="translations">
	{(t) => (
	<div id="hero">
		<ErrorAlert />

        <div className="container">
          <div className="row">
            <div className="col-xs-12">
              <div className="brand">
                <h1>{t('hero.1')}</h1>
				<Trans i18nKey="hero.2"> 
                <h2>
                  Simple. <span className="text-green">Secure</span>.
                  Transparent.
                </h2>
				</Trans>
              </div>
            </div>

            <ExchangeWidget />
            <CoinPrices />
          </div>
        </div>

        <div id="mouse-scroll" onClick={() => scrollToElement('#about')}>
          <span className="arrow-1" aria-hidden="true" />
          <span className="arrow-2" aria-hidden="true" />
          <span className="arrow-3" aria-hidden="true" />
        </div>
      </div>
	)}
	</I18n>
    );
  }
}

export default Hero;
