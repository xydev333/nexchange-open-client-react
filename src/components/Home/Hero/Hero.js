import React, { Component } from 'react';

import ExchangeWidget from './ExchangeWidget/ExchangeWidget';
import CoinPrices from './CoinPrices/CoinPrices';
import ErrorAlert from './ErrorAlert/ErrorAlert';
import Header from 'Components/Header/Header';

import styles from './Hero.scss';

class Hero extends Component {
  render() {
    return (
      <div className={styles.hero}>
        <Header />
        <ErrorAlert />

        <div className="container">
          <div className="row">
            <div className="col-xs-12">
              <div className={styles.brand}>
                <h1>Exchange Cryptocurrencies</h1>
                <h2>
                  Simple. <span className="text-green">Secure</span>. Transparent.
                </h2>
              </div>
            </div>

            <ExchangeWidget store={this.props.store} />
          </div>
        </div>

        <CoinPrices />
      </div>
    );
  }
}

export default Hero;
