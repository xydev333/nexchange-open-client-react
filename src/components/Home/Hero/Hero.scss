.hero {
  position: relative;
  padding: 0 0 75px 0;
  height: 100vh;
  background-color: #01031e;
  background-image: url('../../../img/ellipse.svg');
  background-position: -50% 50%;
  background-size: auto 200%;
  background-repeat: no-repeat;
  text-align: center;
  color: #fff;
}

#hero {
  h2 {
    margin-top: 0;
    font-weight: 400;
    font-size: 26px;
  }
  .brand {
    margin-bottom: 50px;
    padding-top: 50px;
  }

  .alert {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    z-index: 2;
  }

  @media (min-width: $screen-sm-min) {
    padding: 0 0 75px;
    .brand {
      padding-top: 40px;
    }
    .alert {
      position: absolute;
    }
    h2 {
      font-size: 34px;
    }
  }
}
