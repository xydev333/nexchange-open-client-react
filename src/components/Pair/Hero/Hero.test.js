import React from 'react';
import { shallow } from 'enzyme';
import Hero from './Hero';


const setup = ({
  selectedCoin = {
    deposit: 'eth',
    receive: 'usd'
  }
}) => ({
  selectedCoin
})
describe('Hero', () => {
  it('renders correctly', () => {
    const props = setup({})
    const wrapShallow = shallow(<Hero {...props} />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
