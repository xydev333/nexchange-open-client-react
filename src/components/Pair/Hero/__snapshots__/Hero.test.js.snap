// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
  >
    [Function]
  </I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HelmetWrapper
          defer={true}
          encodeSpecialCharacters={true}
        >
          <title>
            PerExchangePairHEAD.title
          </title>
          <meta
            content="PerExchangePairHEAD.meta_description"
            name="description"
          />
        </HelmetWrapper>,
        <div
          className="hero"
        >
          <Connect(ErrorAlert) />
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-12"
              >
                <div
                  className="brand"
                >
                  <h1>
                    PerExchangePairBODY.H1
                  </h1>
                  <Trans
                    i18nKey="hero.2"
                  >
                    <h2>
                      Simple. 
                      <span
                        className="text-green"
                      >
                        Secure
                      </span>
                      . Transparent.
                    </h2>
                  </Trans>
                </div>
              </div>
              <div
                className="widget"
              >
                <Connect(OrderBookWidget)
                  selectedCoin={
                    Object {
                      "deposit": "eth",
                      "receive": "usd",
                    }
                  }
                />
              </div>
            </div>
          </div>
          <Connect(CoinPrices) />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <title>
              PerExchangePairHEAD.title
            </title>,
            <meta
              content="PerExchangePairHEAD.meta_description"
              name="description"
            />,
          ],
          "defer": true,
          "encodeSpecialCharacters": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PerExchangePairHEAD.title",
            },
            "ref": null,
            "rendered": "PerExchangePairHEAD.title",
            "type": "title",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "content": "PerExchangePairHEAD.meta_description",
              "name": "description",
            },
            "ref": null,
            "rendered": null,
            "type": "meta",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(ErrorAlert) />,
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-12"
                >
                  <div
                    className="brand"
                  >
                    <h1>
                      PerExchangePairBODY.H1
                    </h1>
                    <Trans
                      i18nKey="hero.2"
                    >
                      <h2>
                        Simple. 
                        <span
                          className="text-green"
                        >
                          Secure
                        </span>
                        . Transparent.
                      </h2>
                    </Trans>
                  </div>
                </div>
                <div
                  className="widget"
                >
                  <Connect(OrderBookWidget)
                    selectedCoin={
                      Object {
                        "deposit": "eth",
                        "receive": "usd",
                      }
                    }
                  />
                </div>
              </div>
            </div>,
            <Connect(CoinPrices) />,
          ],
          "className": "hero",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <div
                  className="col-xs-12"
                >
                  <div
                    className="brand"
                  >
                    <h1>
                      PerExchangePairBODY.H1
                    </h1>
                    <Trans
                      i18nKey="hero.2"
                    >
                      <h2>
                        Simple. 
                        <span
                          className="text-green"
                        >
                          Secure
                        </span>
                        . Transparent.
                      </h2>
                    </Trans>
                  </div>
                </div>
                <div
                  className="widget"
                >
                  <Connect(OrderBookWidget)
                    selectedCoin={
                      Object {
                        "deposit": "eth",
                        "receive": "usd",
                      }
                    }
                  />
                </div>
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-xs-12"
                  >
                    <div
                      className="brand"
                    >
                      <h1>
                        PerExchangePairBODY.H1
                      </h1>
                      <Trans
                        i18nKey="hero.2"
                      >
                        <h2>
                          Simple. 
                          <span
                            className="text-green"
                          >
                            Secure
                          </span>
                          . Transparent.
                        </h2>
                      </Trans>
                    </div>
                  </div>,
                  <div
                    className="widget"
                  >
                    <Connect(OrderBookWidget)
                      selectedCoin={
                        Object {
                          "deposit": "eth",
                          "receive": "usd",
                        }
                      }
                    />
                  </div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="brand"
                    >
                      <h1>
                        PerExchangePairBODY.H1
                      </h1>
                      <Trans
                        i18nKey="hero.2"
                      >
                        <h2>
                          Simple. 
                          <span
                            className="text-green"
                          >
                            Secure
                          </span>
                          . Transparent.
                        </h2>
                      </Trans>
                    </div>,
                    "className": "col-xs-12",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <h1>
                          PerExchangePairBODY.H1
                        </h1>,
                        <Trans
                          i18nKey="hero.2"
                        >
                          <h2>
                            Simple. 
                            <span
                              className="text-green"
                            >
                              Secure
                            </span>
                            . Transparent.
                          </h2>
                        </Trans>,
                      ],
                      "className": "brand",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "PerExchangePairBODY.H1",
                        },
                        "ref": null,
                        "rendered": "PerExchangePairBODY.H1",
                        "type": "h1",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "children": <h2>
                            Simple. 
                            <span
                              className="text-green"
                            >
                              Secure
                            </span>
                            . Transparent.
                          </h2>,
                          "i18nKey": "hero.2",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": Array [
                              "Simple. ",
                              <span
                                className="text-green"
                              >
                                Secure
                              </span>,
                              ". Transparent.",
                            ],
                          },
                          "ref": null,
                          "rendered": Array [
                            "Simple. ",
                            Object {
                              "instance": null,
                              "key": undefined,
                              "nodeType": "host",
                              "props": Object {
                                "children": "Secure",
                                "className": "text-green",
                              },
                              "ref": null,
                              "rendered": "Secure",
                              "type": "span",
                            },
                            ". Transparent.",
                          ],
                          "type": "h2",
                        },
                        "type": [Function],
                      },
                    ],
                    "type": "div",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(OrderBookWidget)
                      selectedCoin={
                        Object {
                          "deposit": "eth",
                          "receive": "usd",
                        }
                      }
                    />,
                    "className": "widget",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "selectedCoin": Object {
                        "deposit": "eth",
                        "receive": "usd",
                      },
                      "store": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
          >
            <title>
              PerExchangePairHEAD.title
            </title>
            <meta
              content="PerExchangePairHEAD.meta_description"
              name="description"
            />
          </HelmetWrapper>,
          <div
            className="hero"
          >
            <Connect(ErrorAlert) />
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-xs-12"
                >
                  <div
                    className="brand"
                  >
                    <h1>
                      PerExchangePairBODY.H1
                    </h1>
                    <Trans
                      i18nKey="hero.2"
                    >
                      <h2>
                        Simple. 
                        <span
                          className="text-green"
                        >
                          Secure
                        </span>
                        . Transparent.
                      </h2>
                    </Trans>
                  </div>
                </div>
                <div
                  className="widget"
                >
                  <Connect(OrderBookWidget)
                    selectedCoin={
                      Object {
                        "deposit": "eth",
                        "receive": "usd",
                      }
                    }
                  />
                </div>
              </div>
            </div>
            <Connect(CoinPrices) />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <title>
                PerExchangePairHEAD.title
              </title>,
              <meta
                content="PerExchangePairHEAD.meta_description"
                name="description"
              />,
            ],
            "defer": true,
            "encodeSpecialCharacters": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PerExchangePairHEAD.title",
              },
              "ref": null,
              "rendered": "PerExchangePairHEAD.title",
              "type": "title",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "content": "PerExchangePairHEAD.meta_description",
                "name": "description",
              },
              "ref": null,
              "rendered": null,
              "type": "meta",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(ErrorAlert) />,
              <div
                className="container"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-xs-12"
                  >
                    <div
                      className="brand"
                    >
                      <h1>
                        PerExchangePairBODY.H1
                      </h1>
                      <Trans
                        i18nKey="hero.2"
                      >
                        <h2>
                          Simple. 
                          <span
                            className="text-green"
                          >
                            Secure
                          </span>
                          . Transparent.
                        </h2>
                      </Trans>
                    </div>
                  </div>
                  <div
                    className="widget"
                  >
                    <Connect(OrderBookWidget)
                      selectedCoin={
                        Object {
                          "deposit": "eth",
                          "receive": "usd",
                        }
                      }
                    />
                  </div>
                </div>
              </div>,
              <Connect(CoinPrices) />,
            ],
            "className": "hero",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="row"
                >
                  <div
                    className="col-xs-12"
                  >
                    <div
                      className="brand"
                    >
                      <h1>
                        PerExchangePairBODY.H1
                      </h1>
                      <Trans
                        i18nKey="hero.2"
                      >
                        <h2>
                          Simple. 
                          <span
                            className="text-green"
                          >
                            Secure
                          </span>
                          . Transparent.
                        </h2>
                      </Trans>
                    </div>
                  </div>
                  <div
                    className="widget"
                  >
                    <Connect(OrderBookWidget)
                      selectedCoin={
                        Object {
                          "deposit": "eth",
                          "receive": "usd",
                        }
                      }
                    />
                  </div>
                </div>,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col-xs-12"
                    >
                      <div
                        className="brand"
                      >
                        <h1>
                          PerExchangePairBODY.H1
                        </h1>
                        <Trans
                          i18nKey="hero.2"
                        >
                          <h2>
                            Simple. 
                            <span
                              className="text-green"
                            >
                              Secure
                            </span>
                            . Transparent.
                          </h2>
                        </Trans>
                      </div>
                    </div>,
                    <div
                      className="widget"
                    >
                      <Connect(OrderBookWidget)
                        selectedCoin={
                          Object {
                            "deposit": "eth",
                            "receive": "usd",
                          }
                        }
                      />
                    </div>,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <div
                        className="brand"
                      >
                        <h1>
                          PerExchangePairBODY.H1
                        </h1>
                        <Trans
                          i18nKey="hero.2"
                        >
                          <h2>
                            Simple. 
                            <span
                              className="text-green"
                            >
                              Secure
                            </span>
                            . Transparent.
                          </h2>
                        </Trans>
                      </div>,
                      "className": "col-xs-12",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <h1>
                            PerExchangePairBODY.H1
                          </h1>,
                          <Trans
                            i18nKey="hero.2"
                          >
                            <h2>
                              Simple. 
                              <span
                                className="text-green"
                              >
                                Secure
                              </span>
                              . Transparent.
                            </h2>
                          </Trans>,
                        ],
                        "className": "brand",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "PerExchangePairBODY.H1",
                          },
                          "ref": null,
                          "rendered": "PerExchangePairBODY.H1",
                          "type": "h1",
                        },
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "children": <h2>
                              Simple. 
                              <span
                                className="text-green"
                              >
                                Secure
                              </span>
                              . Transparent.
                            </h2>,
                            "i18nKey": "hero.2",
                          },
                          "ref": null,
                          "rendered": Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": Array [
                                "Simple. ",
                                <span
                                  className="text-green"
                                >
                                  Secure
                                </span>,
                                ". Transparent.",
                              ],
                            },
                            "ref": null,
                            "rendered": Array [
                              "Simple. ",
                              Object {
                                "instance": null,
                                "key": undefined,
                                "nodeType": "host",
                                "props": Object {
                                  "children": "Secure",
                                  "className": "text-green",
                                },
                                "ref": null,
                                "rendered": "Secure",
                                "type": "span",
                              },
                              ". Transparent.",
                            ],
                            "type": "h2",
                          },
                          "type": [Function],
                        },
                      ],
                      "type": "div",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Connect(OrderBookWidget)
                        selectedCoin={
                          Object {
                            "deposit": "eth",
                            "receive": "usd",
                          }
                        }
                      />,
                      "className": "widget",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "selectedCoin": Object {
                          "deposit": "eth",
                          "receive": "usd",
                        },
                        "store": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
    Symbol(enzyme.__providerValues__): Map {},
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
